crawl-ref/source/player.cc

```diff
@@ -4745,11 +4745,21 @@ void dec_channel_player(int delay)
 bool invis_allowed(bool quiet, string *fail_reason)
 {
     string msg;
+    string reason = "";
     bool success = true;
 
     if (you.haloed() && you.halo_radius() != -1)
     {
-        msg = "Your halo prevents invisibility.";
+        if (player_equip_unrand(UNRAND_EOS))
+            reason = "weapon";
+        if (you.attribute[ATTR_HEAVENLY_STORM] > 0 ||
+            you.religion == GOD_SHINING_ONE)
+        {
+            reason = (reason == "" ? "Divine Halo" : reason + " and Divine Halo");
```

Comment:
`.empty()`.

Reply:
Also I think capitalising it is wrong. It's not an ability named "Divine Halo" (right?), it's a halo which is divine.